operator-common/src/main/java/io/strimzi/operator/common/AbstractOperator.java

```diff
@@ -488,31 +488,38 @@ public AtomicInteger getResourceCounter() {
      * @param reconciliation reconciliation to use to update the resource state metric
      * @param ready if reconcile was successful and the resource is ready
      */
-    private void updateResourceState(Reconciliation reconciliation, boolean ready) {
-        Tags metricTags = Tags.of(
+    private void updateResourceState(Reconciliation reconciliation, boolean ready, Throwable cause) {
+        String key = reconciliation.namespace() + ":" + reconciliation.kind() + "/" + reconciliation.name();
+
+        Tags metricTagsWithReason = Tags.of(
                 Tag.of("kind", reconciliation.kind()),
                 Tag.of("name", reconciliation.name()),
-                Tag.of("resource-namespace", reconciliation.namespace()));
+                Tag.of("resource-namespace", reconciliation.namespace()),
+                Tag.of("failure-reason", cause == null ? "none" : cause.getMessage() == null ? "unknown error" : cause.getMessage()));
 
         T cr = resourceOperator.get(reconciliation.namespace(), reconciliation.name());
```

Comment:
Do we really need a synchronous get here just to check if the resource is not null? Can't it be derived from the reconciliation in some other way?